src/Domain/CampaignStatisticsRepository.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace MatchBot\Domain;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+
+class CampaignStatisticsRepository
+{
+    /** @var EntityRepository<CampaignStatistics>  */
+    private EntityRepository $doctrineRepository;
+
+    /**
+     * @psalm-suppress PossiblyUnusedMethod Called by DI container
+     */
+    public function __construct(
+        private EntityManagerInterface $em,
+        private MatchFundsService $matchFundsService,
+    ) {
+        $this->doctrineRepository = $em->getRepository(CampaignStatistics::class);
+    }
+
+    /**
+     * Creates or finds + updates a {@see CampaignStatistics} record; doesn't flush, so callers need to when done
+     * building stats.
+     */
+    public function updateStatistics(
+        Campaign $campaign,
+        Money $donationSum,
+        Money $amountRaised,
+        Money $matchFundsUsed,
+    ): void {
+        $statistics = $this->doctrineRepository->findOneBy(['campaign' => $campaign]);
+
+        if ($statistics) {
+            $statistics->setAmountRaised($amountRaised);
+            $statistics->setMatchFundsUsed($matchFundsUsed);
```

Comment:
this looks like its missing a line to set the `donationSum` . Also I would use a single function on stats to set three stats at once because then it can enforce the invariant internally. Something like this:


```suggestion
            $statistics->setDonationTotals(amountRaised: $amountRaised, matchFundsUsed: $matchFundsUsed, donationSum: $donationSum);
```

Reply:
Good catch, thanks! Liking the idea of the combined setter, will do that shortly.